backend/src/db/schemas/external-group-org-role-mappings.ts

```diff
@@ -0,0 +1,27 @@
+// Code generated by automation script, DO NOT EDIT.
+// Automated by pulling database and generating zod schema
+// To update. Just run npm run generate:schema
+// Written by akhilmhdh.
+
+import { z } from "zod";
+
+import { TImmutableDBKeys } from "./models";
+
+export const ExternalGroupOrgRoleMappingsSchema = z.object({
+  id: z.string().uuid(),
+  groupName: z.string(),
+  role: z.string(),
+  roleId: z.string().uuid().nullable().optional(),
```

Comment:
why is roleId here nullable and optional? it should be required right?

Reply:
`roleId` is only used if `role` is `custom` - if using a reserved role, ie `admin`, `member` etc. `roleId` is `null` (following `org_memberships` table convention)